arraycontext/impl/pytato/__init__.py

```diff
@@ -260,26 +260,37 @@ def __init__(
         self.using_svm = None
 
         if allocator is None:
+            import sys
+
             from pyopencl.characterize import has_coarse_grain_buffer_svm
             has_svm = has_coarse_grain_buffer_svm(queue.device)
-            if has_svm:
-                self.using_svm = True
+            is_nvidia_cl_with_pytest = (queue.device.platform.name == "NVIDIA CUDA"
+                                        and "pytest" in sys.modules)
```

Comment:
Shouldn't this logic be in one of the `PytestArrayContextFactory`? It feels weird to put it in here.

Reply:
I agree that that's probably better.

Reply:
What do you think of https://github.com/inducer/arraycontext/pull/200/commits/618f16a20c8aad0a8a89f3a306f8aba05fb6d8ee ?

Reply:
Much nicer!